babel/numbers.py

```diff
@@ -774,6 +774,8 @@ def format_currency(
                                           locale=locale, currency_digits=currency_digits,
                                           decimal_quantization=decimal_quantization, group_separator=group_separator,
                                           numbering_system=numbering_system)
+    if locale is None:
+        locale = default_locale()
```

Comment:
It bugs me that `default_locale()` determines the locale as a string, parses it to a `Locale`, and converts it to its string identifier. Then immediately after, it's parsed back to a `Locale`.

But I can't think of a good way around this, other than a breaking change in `default_locale()`.